public/controllers/management/components/management/common/actions-buttons.tsx

```diff
@@ -0,0 +1,186 @@
+import { resourceDictionary, ResourcesHandler } from './resources-handler';
+import { WzButtonPermissions } from '../../../../../components/common/permissions/button';
+
+import { UI_ERROR_SEVERITIES } from '../../../../../react-services/error-orchestrator/types';
+import { UI_LOGGER_LEVELS } from '../../../../../../common/constants';
+import { getErrorOrchestrator } from '../../../../../react-services/common-services';
+
+/**
+ * Uploads the files
+ * @param {Array} files
+ * @param {String} path
+ */
+async function uploadFiles(files, resource, overwrite) {
+  try {
+    let errors = false;
+    let results: any[] = [];
+    const resourcesHandler = new ResourcesHandler(resource);
+    for (let idx in files) {
+      const { file, content } = files[idx];
+      try {
+        await resourcesHandler.updateFile(file, content, overwrite);
+        results.push({
+          index: idx,
+          uploaded: true,
+          file: file,
+          error: 0,
+        });
+      } catch (error) {
+        errors = true;
+        results.push({
+          index: idx,
+          uploaded: false,
+          file: file,
+          error: error,
+        });
+      }
+    }
+    if (errors) throw results;
+    return results;
+  } catch (error) {
+    throw error;
+  }
+}
+
+const getReadPermissionsFiles = (section) => {
+  const { permissionResource } = resourceDictionary[section];
+  return [
+    {
+      action: `${section}:read`,
+      resource: permissionResource('*'),
+    },
+  ];
+};
+
+const getUpdatePermissionsFiles = (section) => {
+  const { permissionResource } = resourceDictionary[section];
+  return [
+    {
+      action: `${section}:update`,
+      resource: permissionResource('*'),
+    },
+    {
+      action: `${section}:read`,
+      resource: permissionResource('*'),
+    },
+  ];
+};
+
+// Add new rule button
+export const AddNewFileButton = ({ section, updateAddingFile }) => (
+  <>{
+    section !== 'lists' &&
+    <WzButtonPermissions
+      permissions={getUpdatePermissionsFiles(section)}
+      buttonType="empty"
+      iconType="plusInCircle"
+      onClick={() =>
+        updateAddingFile({
+          name: '',
+          content: '<!-- Modify it at your will. -->',
+          path: `etc/${section}`,
+        })
+      }
+    >
+      {`Add new ${section} file`}
+    </WzButtonPermissions>
+  }</>
+)
+
+//Add new CDB list button
+export const AddNewCdbListButton = (({ section, updateListContent }) => {
+  return <>
+      <WzButtonPermissions
+        buttonType="empty"
+        permissions={getUpdatePermissionsFiles(section)}
+        iconType="plusInCircle"
+        onClick={() =>
+          updateListContent({
+            name: false,
+            content: '',
+            path: 'etc/lists',
+          })
+        }
+      >
+        {`Add new ${section} file`}
+      </WzButtonPermissions>
+  </>
+});
+
+// Manage files
+export const ManageFiles = (({ section, showingFiles, ...props }) => {
```

Comment:
[Naming convention](https://github.com/wazuh/wazuh-kibana-app/blob/master/STYLEGUIDE.md#use-lowercamelcase-for-variables-properties-and-function-names)

Reply:
These are _React function components_ that follow the same convention used in the documentation:
https://reactjs.org/docs/components-and-props.html